plugins/defaultplugins/aclplugin/vppcalls/acl_vppcalls_test.go

```diff
@@ -0,0 +1,501 @@
+					},
+					IcmpTypeRange: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp_Range{
+						First: 1150,
+						Last:  1250,
+					},
+				},
+			},
+		},
+	},
+	{
+		RuleName: "denyICMPv6",
+		AclAction: acl.AclAction_DENY,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			IpRule: &acl.AccessLists_Acl_Rule_Match_IpRule{
+				Icmp: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp{
+					Icmpv6: true,
+					IcmpCodeRange: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp_Range{
+						First: 150,
+						Last:  250,
+					},
+					IcmpTypeRange: &acl.AccessLists_Acl_Rule_Match_IpRule_Icmp_Range{
+						First: 1150,
+						Last:  1250,
+					},
+				},
+			},
+		},
+	},
+	{
+		RuleName: "permitTCP",
+		AclAction: acl.AclAction_PERMIT,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			IpRule: &acl.AccessLists_Acl_Rule_Match_IpRule{
+				Tcp: &acl.AccessLists_Acl_Rule_Match_IpRule_Tcp{
+					TcpFlagsMask:  20,
+					TcpFlagsValue: 10,
+					SourcePortRange: &acl.AccessLists_Acl_Rule_Match_IpRule_PortRange{
+						LowerPort: 150,
+						UpperPort: 250,
+					},
+					DestinationPortRange: &acl.AccessLists_Acl_Rule_Match_IpRule_PortRange{
+						LowerPort: 1150,
+						UpperPort: 1250,
+					},
+				},
+			},
+		},
+	},
+	{
+		RuleName: "denyUDP",
+		AclAction: acl.AclAction_DENY,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			IpRule: &acl.AccessLists_Acl_Rule_Match_IpRule{
+				Udp: &acl.AccessLists_Acl_Rule_Match_IpRule_Udp{
+					SourcePortRange: &acl.AccessLists_Acl_Rule_Match_IpRule_PortRange{
+						LowerPort: 150,
+						UpperPort: 250,
+					},
+					DestinationPortRange: &acl.AccessLists_Acl_Rule_Match_IpRule_PortRange{
+						LowerPort: 1150,
+						UpperPort: 1250,
+					},
+				},
+			},
+		},
+	},
+}
+
+var acl_MACIPrules = []*acl.AccessLists_Acl_Rule {
+	{
+		RuleName: "denyIPv4",
+		AclAction: acl.AclAction_DENY,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			MacipRule: &acl.AccessLists_Acl_Rule_Match_MacIpRule{
+				SourceAddress: "192.168.0.1",
+				SourceAddressPrefix: uint32(16),
+				SourceMacAddress: "11:44:0A:B8:4A:35",
+				SourceMacAddressMask: "ff:ff:ff:ff:00:00",
+			},
+		},
+	},
+	{
+		RuleName: "denyIPv6",
+		AclAction: acl.AclAction_DENY,
+		Match: &acl.AccessLists_Acl_Rule_Match{
+			MacipRule: &acl.AccessLists_Acl_Rule_Match_MacIpRule{
+				SourceAddress: "dead::1",
+				SourceAddressPrefix: uint32(64),
+				SourceMacAddress: "11:44:0A:B8:4A:35",
+				SourceMacAddressMask: "ff:ff:ff:ff:00:00",
+			},
+		},
+	},
+}
+
+func TestAddIPAcl(t *testing.T) {
+	ctx := vppcallmock.SetupTestCtx(t)
+	defer ctx.TeardownTestCtx()
+	ctx.MockVpp.MockReply(&acl_api.ACLAddReplaceReply{})
+
+	aclIndex, err := AddIPAcl( acl_IPrules, "test0", logrus.DefaultLogger(),  ctx.MockChannel, nil)
+	Expect(err).To(BeNil())
+	Expect(aclIndex).To(BeEquivalentTo(0))
+
+	_, err = AddIPAcl( acl_NOrules, "test1", logrus.DefaultLogger(),  ctx.MockChannel, nil)
+	Expect(err).To(Not(BeNil()))
+	Expect(err.Error()).To(BeEquivalentTo("no rules found for ACL test1"))
```

Comment:
I think you do not need to compare the full error string. We'd have to change tests even after changing log message.  Test, whether an error occurred, is enough.

@ondrej-fabry what do you think?

Reply:
DONE